### PR TITLE
[IMPROVE] Replace remaing discussion creation modals with React modal.

### PR DESCRIPTION
**Pull request by gabriellsh**
### before
image

### after
image

## Steps to test or reproduce

Try to create a discution from a message and from the message box (message editor plus sign)
When creating from a message, the title of the discution should already be filled with part of the message.
In both cases, the "parent room' field should be disabled, and the current room name should be displayed there.

### CODE REVIEW

File: client/contexts/ServerContext/endpoints.ts [hunk 1]
```diff
@@ -54,6 +55,7 @@ export type ServerEndpoints = {
 	'livechat/tags.list': LivechatTagsList;
 	'livechat/department': LivechatDepartment;
 	'livechat/departments.by-unit/': LivechatDepartmentsByUnit;
+	'rooms.info': RoomsInfoEndpoint;
```
Comment (dougfabris) [inline client/contexts/ServerContext/endpoints.ts:58]: shouldn't we use it in singular `room.info: RoomInfoEndpoint`?

File: client/components/CreateDiscussion/DefaultParentRoom.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { Skeleton, Select, Callout } from '@rocket.chat/fuselage';
+import React, { useMemo, ReactElement } from 'react';
+
+import { useTranslation } from '../../contexts/TranslationContext';
+import { AsyncStatePhase } from '../../hooks/useAsyncState';
+import { useEndpointData } from '../../hooks/useEndpointData';
+
+const DefaultParentRoom = ({ defaultParentRoom }: { defaultParentRoom: string }): ReactElement => {
```
Comment (dougfabris) [inline client/components/CreateDiscussion/DefaultParentRoom.tsx:8]: Isn't `DefaultParentRoomField` more descriptive?

File: client/components/CreateDiscussion/CreateDiscussion.tsx [hunk 3]
```diff
@@ -30,22 +32,22 @@ type CreateDiscussionFormValues = {
...
+	onClose,
+	defaultParentRoom,
+	parentMessageId,
+	nameSuggestion,
+}: CreateDiscussionProps): ReactElement => {
 	const t = useTranslation();
 
 	const { values, handlers } = useForm({
-		name: '',
+		name: nameSuggestion || '',
```
Comment (dougfabris) [inline client/components/CreateDiscussion/CreateDiscussion.tsx:50]: If the message is a quote, we receive additionally the link as a suggestion:
image

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: client/components/CreateDiscussion/CreateDiscussion.tsx (hunk 3 above)
Comment (gabriellsh) [reply client/components/CreateDiscussion/CreateDiscussion.tsx:50]: This is the same behavior of the old modal. I agree it's not pretty, but there could be anything on the message, we just take the raw text and clip it to a set character count.

**Review by dougfabris — APPROVED** [review]

**Pull request merged**